saleor/checkout/base_calculations.py

```diff
@@ -182,12 +182,15 @@ def calculate_base_price_for_shipping_method(
 def base_checkout_total(
     subtotal: TaxedMoney,
     shipping_price: TaxedMoney,
-    discount: TaxedMoney,
+    discount: Money,
     currency: str,
 ) -> TaxedMoney:
     """Return the total cost of the checkout."""
+    zero = zero_taxed_money(currency)
     total = subtotal + shipping_price - discount
-    return max(total, zero_taxed_money(currency))
+    if total.gross <= zero.gross:
```

Comment:
Can we add an explanation here? For someone not familiar with the issue, this could not be clear.

Reply:
added.